File: uit.hotel.client/graphql/documents/price.ts
import gql from 'graphql-tag';

export const getPrices = gql`
    query getPrices {
        prices {
            id
            hourPrice
            nightPrice
            dayPrice
            effectiveStartDate
            createDate
            roomKind {
                id
                name
            }
        }
    }
`;

export const getPrice = gql`
    query getPrice($id: ID!) {
        price(id: $id) {
            id
            createDate
            effectiveStartDate
            earlyCheckInFee
            lateCheckOutFee
            hourPrice
            nightPrice
            dayPrice
            weekPrice
            monthPrice
            roomKind {
                id
                name
            }
        }
    }
`;

export const createPrice = gql`
    mutation createPrice($input: PriceCreateInput!) {
        createPrice(input: $input) {
            id
        }
    }
`;

File: uit.hotel/Models/Price.cs
using System;
using System.Linq;
using Realms;

namespace uit.hotel.Models
{
    public class Price : RealmObject
    {
        [PrimaryKey]
        public int Id { get; set; }
        public long HourPrice { get; set; }
        public long DayPrice { get; set; }
        public long NightPrice { get; set; }
        public long WeekPrice { get; set; }
        public long MonthPrice { get; set; }
        public long LateCheckOutFee { get; set; }
        public long EarlyCheckInFee { get; set; }
        [Indexed]
        public DateTimeOffset EffectiveStartDate { get; set; }
        public DateTimeOffset CreateDate { get; set; }
        public Employee Employee { get; set; }
        public RoomKind RoomKind { get; set; }
    }
}
